File: src/DomainF/Teams/Rules.fs
module Orleans.Tournament.Domain.Teams.Rules
open Constants

let TeamExists (teamState : TeamState) =
    match teamState.Created with
    | true -> Ok teamState
    | false -> Error BusinessErrors.TeamDoesNotExist

let TeamDoesNotExists (teamState : TeamState) =
    match teamState.Created with
    | false -> Ok teamState
    | true -> Error BusinessErrors.TeamAlreadyExist

File: src/DomainC/Teams/Rules.cs
namespace Tournament.Domain.Teams;

public partial class TeamState
{
    public Results TeamExists()
        => Created
            ? Results.Unit
            : Results.TeamDoesNotExist;

    public Results TeamDoesNotExists()
        => !Created
            ? Results.Unit
            : Results.TeamAlreadyExist;
}